reader-osm/src/test/java/com/graphhopper/reader/osm/OSMReaderTest.java

```diff
@@ -411,14 +414,31 @@ public void avoidsLoopEdges() {
         while (iter.next()) {
             assertTrue("found a loop", iter.getAdjNode() != iter.getBaseNode());
         }
-        int nodeB = -1;
-        for (int i = 0; i < graph.getNodes(); i++) {
-            if (Math.abs(graph.getNodeAccess().getLat(i) - 12) < 1.e-3) {
-                nodeB = i;
-            }
+        int nodeB = AbstractGraphStorageTester.getIdOf(graph, 12);
+        assertTrue("could not find OSM node B", nodeB > -1);
+        assertEquals(4, GHUtility.count(graph.createEdgeExplorer().setBaseNode(nodeB)));
```

Comment:
Here I'm using now the edge counting as the getNeightbors method uses a set which returns 3 instead of the more correct 4. (changing getNeighbors to using a list even with a Collections.sort(list) would make several tests fail and is not worth the effort IMO)

Reply:
Maybe the best would be to check both ? Three neighbors, four edges ? But sure for the nodeB to nodeB loop counting edges is better.